### ISSUE TITLE
YAPF reformats its own output

### ISSUE THREAD

**Issue by eddie-dunn** [google/yapf#372]
Let's say I start with this Pylint and PEP8 compatible piece of code:

```python
"""foo"""
import urllib.parse as urllib


def func():
    """func"""
    value = 1
    url = "http://{0}/axis-cgi/admin/param.cgi?{1}".format(
        value, urllib.urlencode({'action': 'update', 'parameter': value}))
    return url
```

It looks fine, but running it through yapf produces this:

```python
"""foo"""
import urllib.parse as urllib


def func():
    """func"""
    value = 1
    url = "http://{0}/cgi/admin/param.cgi?{1}".format(
        value, urllib.urlencode({
            'action': 'update',
            'parameter': value
        }))
    return url
```

No biggie, I get the dict key-value pairs on separate lines, which is arguably an improvement at the cost of some LOCs.

However, if I run that output through yapf again I get this:

```python
"""foo"""
import urllib.parse as urllib


def func():
    """func"""
    value = 1
    url = "http://{0}/cgi/admin/param.cgi?{1}".format(value,
                                                      urllib.urlencode({
                                                          'action':
                                                          'update',
                                                          'parameter':
                                                          value
                                                      }))
    return url
```

First of all it looks horrible. Second, why would yapf change its  generated output from the first pass, which was perfectly fine?

```bash
$ yapf --version
yapf 0.16.0
```

**Comment by gwelymernans** [comment]
Agreed that it looks horrible. This is probably one of the situations where formatting is based on the original formatting. I'll see what's up.

**Comment by gwelymernans** [comment]
Fixed: 313c65a146692826cba4b1e6768f7af714e6d352
